### PR TITLE
Fix compile-time warnings and other nits

### PR DESCRIPTION
**Pull request by jrabinow**
Fix compile-time warnings and other nits

- prevent incorrect typecasting (comparison between signed and unsigned)
- prevent buffer overflows by replacing sprintf with snprintf
- remove unused variables
- newline added in final output
- add *.swp to gitignore
- whitespace

### CODE REVIEW

**Comment by narugit** [comment]
Hi @jrabinow 
Thank you for creating PR!
I'll check.

**Comment by narugit** [comment]
### Before
```console
$ sudo make 2>&1 | grep "warning"
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
smctemp.cc:61:1: warning: 'OSSpinLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:153:3: warning: 'IOMasterPort' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
smctemp.cc:214:3: warning: 'OSSpinLockLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock_lock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:242:3: warning: 'OSSpinLockUnlock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock_unlock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:326:3: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
smctemp.cc:326:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
6 warnings generated.
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
smctemp_string.cc:20:3: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
1 warning generated.
main.cc:26:26: warning: unused variable 'key' [-Wunused-variable]
main.cc:27:26: warning: unused variable 'val' [-Wunused-variable]
2 warnings generated.
```

### After
```console
$ sudo make 2>&1 | grep "warning"
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
smctemp.cc:61:1: warning: 'OSSpinLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:153:3: warning: 'IOMasterPort' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
smctemp.cc:214:3: warning: 'OSSpinLockLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock_lock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:242:3: warning: 'OSSpinLockUnlock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock_unlock() from <os/lock.h> instead [-Wdeprecated-declarations]
smctemp.cc:326:24: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
5 warnings generated.
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
```

File: main.cc [hunk 1]
```diff
@@ -23,8 +23,6 @@ int main(int argc, char *argv[]) {
 
   kern_return_t result;
   int           op = smctemp::kOpNone;
-  smctemp::UInt32Char_t  key = { 0 };
-  smctemp::SmcVal_t      val;
```
Comment (narugit) [inline main.cc:27]: note: 
```console
main.cc:26:26: warning: unused variable 'key' [-Wunused-variable]
main.cc:27:26: warning: unused variable 'val' [-Wunused-variable]
```

File: smctemp.cc [hunk 2]
```diff
@@ -317,37 +317,37 @@ kern_return_t SmcAccessor::ReadSmcVal(const UInt32Char_t key, SmcVal_t& val) {
...
   SmcKeyData_t  inputStructure;
   SmcKeyData_t  outputStructure;
-    
+
   memset(&inputStructure, 0, sizeof(SmcKeyData_t));
   memset(&outputStructure, 0, sizeof(SmcKeyData_t));
   memset(&val, 0, sizeof(SmcVal_t));
 
   inputStructure.key = string_util::strtoul(key, 4, 16);
-  sprintf(val.key, key);
```
Comment (narugit) [inline smctemp.cc:326]: note:
```console
smctemp.cc:326:3: warning: 'sprintf' is deprecated: 
This function is provided for compatibility reasons only.  
Due to security concerns inherent in the design of sprintf(3), 
it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
```

**Review by narugit — APPROVED** [review]
LGTM

File: smctemp.cc [hunk 3]
```diff
@@ -317,37 +317,37 @@ kern_return_t SmcAccessor::ReadSmcVal(const UInt32Char_t key, SmcVal_t& val) {
...
-    
+  snprintf(val.key, 5, key);
```
Comment (narugit) [inline smctemp.cc:326]:
```suggestion
  snprintf(val.key, sizeof(val.key), key);
```
Could you use `sizeof` in second argument?

**Review by narugit — CHANGES_REQUESTED** [review]

File: smctemp.cc (hunk 3 above)
Comment (jrabinow) [reply smctemp.cc]: Done!

**Review by narugit — APPROVED** [review]
Thank you! LGTM

**Pull request merged**